modtools/deep-embark.lua

```diff
@@ -0,0 +1,321 @@
+      table.insert(featureBlocks, i)
+    end
+  end
+  return featureBlocks
+end
+
+function isValidTiletype(tiletype)
+  local tiletype = df.tiletype[tiletype]
+  local tiletypeAttrs = df.tiletype.attrs[tiletype]
+  local material = tiletypeAttrs.material
+  local forbiddenMaterials = {
+    df.tiletype_material.TREE, -- so as not to embark stranded on top of a tree
+    df.tiletype_material.MUSHROOM,
+    df.tiletype_material.FIRE,
+    df.tiletype_material.CAMPFIRE
+  }
+  for _,forbidden in ipairs(forbiddenMaterials) do
+    if material == forbidden then
+      return false
+    end
+  end
+  local shape = tiletypeAttrs.shape
+  local shapeAttrs = df.tiletype_shape.attrs[shape]
+  if shapeAttrs.walkable and shapeAttrs.basic_shape ~= df.tiletype_shape_basic.Open then -- downward ramps are walkable but open; units placed here would fall
+    return true
+  else
+    return false
+  end
+end
+
+function getValidEmbarkTiles(block)
+  local validTiles = {}
+  for xi = 0,15 do
+    for yi = 0,15 do
+      if block.designation[xi][yi].flow_size == 0
+      and isValidTiletype(block.tiletype[xi][yi]) then
+        table.insert(validTiles, {x = block.map_pos.x + xi, y = block.map_pos.y + yi, z = block.map_pos.z})
+      end
+    end
+  end
+  return validTiles
+end
+
+function reveal(pos, recenter)
+-- creates an unbound glowing barrier at the target location to trigger proper tile revelation when it disappears (fortress mode only)
+-- if recenter is true, shifts the view to the target location
+  local x,y,z = pos2xyz(pos)
+  local block = dfhack.maps.getTileBlock(x,y,z)
+  local tiletype = block.tiletype[x%16][y%16]
+  if tiletype ~= df.tiletype.GlowingBarrier then -- to avoid multiple instances
+    block.tiletype[x%16][y%16] = df.tiletype.GlowingBarrier
+    local barriers = df.global.world.glowing_barriers
+    local barrier = df.glowing_barrier:new()
+    barrier.buildings:insert('#',-1) -- being unbound to a building makes the barrier disappear immediately
+    barrier.pos:assign(pos)
+    barriers:insert('#',barrier)
+    local hfs = df.glowing_barrier:new()
+    hfs.triggered = 1 -- this prevents hfs events (which can otherwise be triggered by the barrier disappearing)
+    barriers:insert('#',hfs)
+    local announcementFlags = df.global.d_init.announcements.flags.ENDGAME_EVENT_1 -- glowing barrier disappearance announcement
+    local oldDO_MEGA = announcementFlags.DO_MEGA
+    announcementFlags.DO_MEGA = false -- turn off popup announcement
+    local oldPAUSE = announcementFlags.PAUSE
+    announcementFlags.PAUSE = false
+    local oldRECENTER = announcementFlags.RECENTER
+    announcementFlags.RECENTER = recenter and true or false
+    local oldA_DISPLAY = announcementFlags.A_DISPLAY
+    announcementFlags.A_DISPLAY = false
+    local oldD_DISPLAY = announcementFlags.D_DISPLAY
+    announcementFlags.D_DISPLAY = recenter and true or false -- won't recenter without an actual announcement
```

Comment:
Looking at https://dfhack.readthedocs.io/en/stable/docs/Lua%20API.html#typed-object-references, I think you could simplify this with:
```lua
local oldFlags = df.global.d_init.announcements.flags.ENDGAME_EVENT_1:new()
...
df.global.d_init.announcements.flags.ENDGAME_EVENT_1:assign(oldFlags)
```

Reply:
A good idea; changed as advised.